crates/ruma-events/src/content.rs

```diff
@@ -9,63 +9,64 @@ use super::{
     RoomAccountDataEventType, StateEventType, ToDeviceEventType,
 };
 
-/// The base trait that all event content types implement.
-///
-/// Use [`macros::EventContent`] to derive this traits. It is not meant to be implemented manually.
-///
-/// [`macros::EventContent`]: super::macros::EventContent
-pub trait EventContent: Sized + Serialize {
-    /// The Rust enum for the event kind's known types.
-    type EventType;
-
-    /// Get the event's type, like `m.room.message`.
-    fn event_type(&self) -> Self::EventType;
-}
-
 /// Extension trait for [`Raw<T>`].
 pub trait RawExt<T: EventContentFromType> {
     /// Try to deserialize the JSON as an event's content with the given event type.
-    fn deserialize_with_type(&self, event_type: T::EventType) -> serde_json::Result<T>;
+    fn deserialize_with_type(&self, event_type: &str) -> serde_json::Result<T>;
```

Comment:
We lose a nice bound here, but I don't think that there is a solution for this.

The only use that I have found of this method in the SDK converts a `String` to an `EventType` to match this signature anyway, so this will avoid a round-trip.